source/WeightedList.h

```diff
@@ -79,22 +96,22 @@ class WeightedList {
 
 private:
 	std::vector<Type> choices;
-	std::vector<std::size_t> weights;
+	std::vector<Weight> weights;
 	std::size_t total = 0;
 };
 
 
 
-template <class T>
-std::size_t erase(WeightedList<T> &list, const T &item)
+template <class T,class W>
+std::size_t erase(WeightedList<T,W> &list, const T &item)
```

Comment:
```suggestion
std::size_t erase(WeightedList<T, W> &list, const T &item)
```
also need to add a space right above, and below multiple times